libmamba/libmambaConfig.cmake.in

```diff
@@ -34,13 +34,20 @@ find_dependency(yaml-cpp)
 if(NOT (TARGET libmamba OR TARGET libmamba-static))
     include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
 
-    if (TARGET libmamba-static)
+    if (TARGET mamba::libmamba-dyn)
+        get_target_property(@PROJECT_NAME@_INCLUDE_DIR libmamba INTERFACE_INCLUDE_DIRECTORIES)
+        get_target_property(@PROJECT_NAME@_LIBRARY libmamba LOCATION)
+    endif()
+
+    if (TARGET mamba::libmamba-static)
         get_target_property(@PROJECT_NAME@_INCLUDE_DIR libmamba-static INTERFACE_INCLUDE_DIRECTORIES)
         get_target_property(@PROJECT_NAME@_STATIC_LIBRARY libmamba-static LOCATION)
-    endif ()
+    endif()
+
+    if(BUILD_SHARED_LIBS AND TARGET mamba::libmamba-dyn)
+        add_library(mamba::libmamba ALIAS mamba::libmamba-dyn)
+    else()
+        add_library(mamba::libmamba ALIAS mamba::libmamba-static)
```

Comment:
I'm a bit skeptical about this alias, I find it confusing to have a target that can be either the dynamic or the static library depending on what has been built and/or installed. The pattern "libname" / "libname-static" for the target names is quite widespread and more explicit.

Reply:
Very often though `libname` would also be either static or dynamic depending on the options, so I thought let's make one target that is always dynlib, and keep the dispatch.

I'm also not such a fan, but it is convenient for inside our own CMake, otherwise we keep repeating if/else in tests/libmambapy/mamba-package...

Reply:
Naive question: is it possible to build both the shared and static libraries, or are both options exclusive?

I am OK about having `mamba::libmamba` alias `mamba::libmamba-dyn`, but I am not sure about having it alias `mamba::libmamba-static`.

Note: I am just getting started on contributing to mamba.

Reply:
> Naive question: it seems possible to build both the shared and static libraries (options seem not exclusive). This might cause a problem if this is actually the case.

Outside Conda it's quite classic to provide both this way.

Note that in this project, we use the static version to build the stand-alone micromamba. I'm ignoring the conda packaging mess in this comment, obviously. :)

Reply:
> Very often though libname would also be either static or dynamic depending on the options, so I thought let's make one target that is always dynlib, and keep the dispatch.

I agree, and I think it's a bad practice that we should fix (at least in conda).

> I'm also not such a fan, but it is convenient for inside our own CMake, otherwise we keep repeating if/else in tests/libmambapy/mamba-package...

Then let's use an alias internally, but I realy think we should not expose it.

Reply:
I think the use case for pubic `mamba::libmamba` is in a VCPKG (or source) like workflow, you configure static/dynamic externally, so then it is not the responsibility of the user CMake script to choose the target.
In fact, that is the bahaviour for CMake `add_library`: it dispatches between dynamic and static.

But really, that works for me too, I don't really care.